packages/components/src/tooltip/test/index.js

```diff
@@ -152,6 +152,40 @@ describe( 'Tooltip', () => {
 			}, TOOLTIP_DELAY );
 		} );
 
+		it( 'should respect custom delay prop when showing popover', () => {
+			const originalMouseEnter = jest.fn();
+			jest.useFakeTimers();
+			const wrapper = mount(
+				<Tooltip text="Help text" delay={ 2000 }>
+					<button
+						onMouseEnter={ originalMouseEnter }
+						onFocus={ originalMouseEnter }
+					>
+						<span>Hover Me!</span>
+					</button>
+				</Tooltip>
+			);
+
+			const button = wrapper.find( 'button' );
+			button.simulate( 'mouseenter', { type: 'mouseenter' } );
+
+			const popoverBeforeTimeout = wrapper.find( 'Popover' );
+			expect( popoverBeforeTimeout ).toHaveLength( 0 );
+			expect( originalMouseEnter ).toHaveBeenCalledTimes( 1 );
+
+			// Popover does not yet exist after default delay, because custom delay is passed.
+			setTimeout( () => {
```

Comment:
Should we utilise [`jest.advanceTimersByTime();`](https://jestjs.io/docs/timer-mocks#advance-timers-by-time) here?

Reply:
This falls into TIL (Today I Learned) category. I'll give this a go in my local and make relevant changes. Thanks.